besu/src/main/java/org/hyperledger/besu/controller/ForkingBesuControllerBuilder.java

```diff
@@ -0,0 +1,335 @@
+      final MiningParameters miningParameters,
+      final SyncState syncState,
+      final EthProtocolManager ethProtocolManager) {
+    return besuControllerBuilders
+        .get(0)
+        .createMiningCoordinator(
+            protocolSchedule,
+            protocolContext,
+            transactionPool,
+            miningParameters,
+            syncState,
+            ethProtocolManager);
+  }
+
+  @Override
+  protected ProtocolSchedule createProtocolSchedule() {
+    return besuControllerBuilders.get(0).createProtocolSchedule();
+  }
+
+  @Override
+  protected ConsensusContext createConsensusContext(
+      final Blockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final ProtocolSchedule protocolSchedule) {
+    return besuControllerBuilders
+        .get(0)
+        .createConsensusContext(blockchain, worldStateArchive, protocolSchedule);
+  }
+
+  @Override
+  protected PluginServiceFactory createAdditionalPluginServices(
+      final Blockchain blockchain, final ProtocolContext protocolContext) {
+    return besuControllerBuilders
+        .get(0)
+        .createAdditionalPluginServices(blockchain, protocolContext);
+  }
+
+  @Override
+  protected JsonRpcMethods createAdditionalJsonRpcMethodFactory(
+      final ProtocolContext protocolContext) {
+    return besuControllerBuilders.get(0).createAdditionalJsonRpcMethodFactory(protocolContext);
+  }
+
+  @Override
+  protected SubProtocolConfiguration createSubProtocolConfiguration(
+      final EthProtocolManager ethProtocolManager) {
+    return besuControllerBuilders.get(0).createSubProtocolConfiguration(ethProtocolManager);
+  }
+
+  @Override
+  protected void validateContext(final ProtocolContext context) {
+    besuControllerBuilders.get(0).validateContext(context);
+  }
+
+  @Override
+  protected String getSupportedProtocol() {
+    return besuControllerBuilders.get(0).getSupportedProtocol();
+  }
+
+  @Override
+  protected EthProtocolManager createEthProtocolManager(
+      final ProtocolContext protocolContext,
+      final boolean fastSyncEnabled,
+      final TransactionPool transactionPool,
+      final EthProtocolConfiguration ethereumWireProtocolConfiguration,
+      final EthPeers ethPeers,
+      final EthContext ethContext,
+      final EthMessages ethMessages,
+      final EthScheduler scheduler,
+      final List<PeerValidator> peerValidators) {
+    return besuControllerBuilders
+        .get(0)
+        .createEthProtocolManager(
+            protocolContext,
+            fastSyncEnabled,
+            transactionPool,
+            ethereumWireProtocolConfiguration,
+            ethPeers,
+            ethContext,
+            ethMessages,
+            scheduler,
+            peerValidators);
+  }
+
+  @Override
+  public BesuControllerBuilder storageProvider(final StorageProvider storageProvider) {
+    besuControllerBuilders.get(0).storageProvider(storageProvider);
+    return super.storageProvider(storageProvider);
```

Comment:
They are not needed. IntelliJ added them and I forgot to change :)

Reply:
> should it be the delegated builder's job to call super rather than this class

This would only initialise some fields in the delegate's builder class, not on the Forking.

I started removing the super's but then realised that we would endup with a lot of duplicated assignments.

Reply:
I had to same problem on my spike branch. This should just return this to satisfy the builder pattern. No need to call super.